### DISCUSSION TITLE
Question about symlinks

### DISCUSSION THREAD

**Discussion by bobbrow** [microsoft/vscode-discussions#122] [Extension Development QnA] [answered]
Hi there,

We have a customer having trouble with one of our extensions and we've tracked it down to some apparent inconsistency in how symlinks are treated.  The customer has a data folder where their project is stored and they also have a symlink from their user folder into that data folder.  When they launch VS Code, they are doing it from their user folder.  From what I can gather in the [bug report](https://github.com/microsoft/vscode-cmake-tools/issues/2713), the workspaceFolder is reported to be either the `realpath` in the data folder or the symlink path in the user folder.

Since I haven't been able to reproduce the behavior (the `realpath` is always the value I get for the workspaceFolder), I just wanted to check and see what the expected behavior is and if there are any known cases where the behavior may change.  We do try to normalize paths in our extension which means that paths will match the Uri's sent by VS Code when they are also using the `realpath`.  If there are no guarantees from VS Code about the workspaceFolder path being resolved, then we might need to adjust some of our logic.

**Comment by isidorn** [comment]
@bpasero can you help for symlink smartness? Thanks!

**Comment by bpasero** [comment, marked as answer]
We preserve symlinks when opening VSCode because that is the intent of the user. So when:

```
code <path to folder that is a symlink>
```

We will not `realpath` the symlink but open it because that was the intent of the user.

We have https://github.com/microsoft/vscode/issues/130082 as a related issue which I believe asks for a way to disable this behaviour.

**Reply by bobbrow** [reply]
Ok, maybe this explains why I wasn't able to repro earlier.  I `cd` into the folder and run `code .`

Here are my observations:
#### This gets `realpath`
```
mkdir -p /tmp/testFolder/Project1
cd ~
ln -s /tmp/testFolder symlink
cd symlink/Project1
code .
```
![image](https://user-images.githubusercontent.com/12818240/188944821-cead6713-d529-4c00-a86a-562cee1e9487.png)


#### This gets symlinks
```
mkdir -p /tmp/testFolder/Project1
cd ~
ln -s /tmp/testFolder symlink
code symlink/Project1
```
![image](https://user-images.githubusercontent.com/12818240/188945469-0d085945-24da-4c67-bd1a-10c4f1018961.png)
